tooling/api/src/main/java/org/jboss/windup/tooling/ToolingModeRunner.java

```diff
@@ -0,0 +1,135 @@
+package org.jboss.windup.tooling;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.rmi.RemoteException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+
+import io.vertx.core.json.JsonObject;
+
+public class ToolingModeRunner implements IProgressMonitorAdapter 
+{
+    private static Logger LOG = Logger.getLogger(ToolingModeRunner.class.getName());
+
+    @Inject
+    private ExecutionBuilder executionBuilder;
+
+    private WindupToolingProgressMonitor progressMonitor;
+
+    public ToolingModeRunner() 
+    {
+        this.progressMonitor = new ProgressMonitorAdapter(this);
+    }
+
+    public void run(
+        Set<String> input,
+        String output,
+        boolean sourceMode,
+        boolean ignoreReport,
+        List<String> ignorePatterns,
+        String windupHome,
+        List<String> source,
+        List<String> target,
+        List<File> customRules) 
+    {
+        try 
+        {
+            executionBuilder.setInput(input);
+            executionBuilder.setOutput(output);
+            executionBuilder.setOption(IOptionKeys.sourceModeOption, sourceMode);
+            executionBuilder.setOption(IOptionKeys.skipReportsRenderingOption, ignoreReport);
+            for (Iterator<String> iter = ignorePatterns.iterator(); iter.hasNext();) 
+            {
+                executionBuilder.ignore(iter.next());
+            }
+            executionBuilder.setWindupHome(windupHome);
+            executionBuilder.setOption(IOptionKeys.sourceOption, source);
+            executionBuilder.setOption(IOptionKeys.targetOption, target);
+            executionBuilder.setOption(IOptionKeys.userRulesDirectoryOption, customRules);
+            executionBuilder.setProgressMonitor((WindupToolingProgressMonitor) this.progressMonitor);
+            ExecutionResults results = executionBuilder.execute();
+            results.serializeToXML(Paths.get(output + File.separatorChar + "results.xml"));
+            System.out.println(":progress: {\"op\":\"complete\"}");
+        } catch (RemoteException e) 
+        {
+            LOG.severe("ToolingModeRunner Error");
+            e.printStackTrace();
+        }
+    }
+
+    private void log(String msg) 
```

Comment:
hummmm, you have defined the logger above, but now this method is outputting to the console the "log" ?

Reply:
There's a logger defined where? All output here is intended to be logged to std output channel.